packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts

```diff
@@ -16,12 +17,14 @@ export class MongoDBDataProvider<Type = any> implements GraphbackDataProvider<Ty
   protected collectionName: string;
   protected tableMap: ModelTableMap;
   protected fieldTransformMap: FieldTransformMap;
+  protected coerceTSFields: boolean;
 
   public constructor(baseType: GraphQLObjectType, db: any) {
     this.db = db;
     this.tableMap = buildModelTableMap(baseType);
     this.collectionName = this.tableMap.tableName;
     this.fieldTransformMap = getFieldTransformations(baseType);
+    this.coerceTSFields = parseMetadata("versioned", baseType)
```

Comment:
Instead of doing this maybe you could define a method `protected coerceVersionFields()`  which for this base class would return false and for the datasync one it returns true.

Reply:
Sorry for late response! Do you think it should always return false for base class? We have the `@versioned` annotation so was wondering if that would become non-functional for plain mongo if we do that?

Reply:
ah right, I thought the `@versioned` was handled by the `DataSyncProvider`. My concern is doing the parsing here which could later impact performance.

Reply:
Hmm fair point about performance, if we do a function for this, I was thinking the parser would be called everytime a query is executed, which kinda seems like a heavier hit wdyt

Reply:
Ideally this information should be carried in a `model: ModelDefinition`. But we do not have that at the moment, so let's keep it this way for now while waiting for https://github.com/aerogear/graphback/pull/1614#issuecomment-651329839 chat